alpha/alpha-proc/src/main/java/org/neo4j/graphalgo/centrality/ParticleFilteringProc.java

```diff
@@ -0,0 +1,135 @@
+import org.neo4j.procedure.*;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class ParticleFilteringProc {
+
+    private static final String DESCRIPTION =
+            "Particle filtering is a 'real time' Personalized Page Rank centrality algorithm";
+
+    @Context
+    public KernelTransaction transaction;
+
+    @Context
+    public Log log;
+
+    @Procedure(value = "gds.alpha.particleFiltering.stream", mode= Mode.READ)
+    @Description(DESCRIPTION)
+    public Stream<CentralityScore> particleFilteringStream(@Name(value = "configuration", defaultValue = "{}") Map<String, Object> configuration){
+        Stream<Map.Entry<Long, Double>> stream = stream(CypherMapWrapper.create(configuration));
+        return stream.map(entry -> new CentralityScore(entry.getKey(), entry.getValue()));
+    }
+
+    private Stream<Map.Entry<Long, Double>> stream(CypherMapWrapper algoConfig) {
+        double numberParticles = algoConfig.getDouble("numberParticles", 10);
+        double minThreshold = algoConfig.getDouble("minimumThreshold", 0.5);
+        double dampingFactor = algoConfig.getDouble("dampingFactor", 0.85);
+        boolean shuffleNeighbors = algoConfig.getBool("shuffleNeighbors", true);
+        List<Node> sourceNodes = CypherMapWrapper.failOnNull("sourceNodes", algoConfig.getChecked("sourceNodes", Collections.emptyList(), List.class));
+
+        String rawRelationshipType = algoConfig.getString("relationshipType", null);
+        RelationshipType relationshipType = rawRelationshipType == null ? null : RelationshipType.withName(rawRelationshipType);
+        Direction direction = Direction.valueOf(algoConfig.getString("direction", "OUTGOING"));
+
+        double tao = 1.0 / numberParticles;
+
+        Map<Long, Double> p = new HashMap<>();
+        Map<Long, Double> v = new HashMap<>();
+        int numberOfSourceNodes = sourceNodes.size();
+        for (Node sourceNode : sourceNodes) {
+            p.put(sourceNode.getId(), ((1.0 / numberOfSourceNodes) * (numberParticles)));
+            v.put(sourceNode.getId(), ((1.0 / numberOfSourceNodes) * (numberParticles)));
+        }
+
+        while (!p.isEmpty()) {
+            Map<Long, Double>  aux = new HashMap<>();
+            for (Long node : p.keySet()) {
+                double particles = p.get(node) * dampingFactor;
+                Node startingNode = transaction.internalTransaction().getNodeById(node);
+
+                List<Long> neighbours = StreamSupport
+                        .stream(getRelationships(startingNode, relationshipType, direction).spliterator(), false)
+                        .map(relationship -> relationship.getOtherNode(startingNode).getId()).collect(Collectors.toList());
+                int neighboursCount = neighbours.size();
+
+                if(shuffleNeighbors) {
```

Comment:
This version of the code is the `unlabelled` version. It is not recommended to use the unlabelled version in large graphs. The `labelled` version should make use of a weighting function on edges.